federation/source-config-handlers/src/main/java/org/codice/ddf/admin/sources/wfs/probe/DiscoverWfsSourceProbeMethod.java

```diff
@@ -13,131 +13,79 @@
+                null,
                 OPTIONAL_FIELDS,
                 SUCCESS_TYPES,
                 FAILURE_TYPES,
                 WARNING_TYPES,
                 RETURN_TYPES);
-        utils = new WfsSourceUtils();
     }
 
     @Override
     public ProbeReport probe(WfsSourceConfiguration configuration) {
-        WfsSourceConfiguration config = new WfsSourceConfiguration(configuration);
-        ProbeReport probeReport = new ProbeReport(buildMessage(SUCCESS_TYPES,
-                FAILURE_TYPES,
-                WARNING_TYPES,
-                endpointIsReachable(config.sourceHostName(), config.sourcePort())));
-        if (probeReport.containsFailureMessages()) {
-            return probeReport;
-        }
-
-        UrlAvailability availability = utils.confirmEndpointUrl(config);
-        if (availability == null) {
-            return probeReport.addMessage(buildMessage(SUCCESS_TYPES,
-                    FAILURE_TYPES,
-                    WARNING_TYPES,
-                    UNKNOWN_ENDPOINT));
-        }
-        probeReport = createProbeReport(SUCCESS_TYPES,
-                FAILURE_TYPES,
-                WARNING_TYPES,
-                availability.getAvailabilityResult());
-        if (probeReport.containsFailureMessages()) {
-            return probeReport;
-        }
-
-        Optional<WfsSourceConfiguration> createdConfig = utils.getPreferredConfig(config);
-        if (!createdConfig.isPresent()) {
-            return probeReport.addMessage(buildMessage(SUCCESS_TYPES,
-                    FAILURE_TYPES,
-                    WARNING_TYPES,
-                    INTERNAL_ERROR));
+        String testUrl = configuration.endpointUrl();
+        if (testUrl == null) {
+            ProbeReport discoverEndpointReport = discoverWfsUrl(
+                    configuration.sourceHostName(), configuration.sourcePort(),
+                    configuration.sourceUserName(), configuration.sourceUserPassword());
+            if (discoverEndpointReport.containsFailureMessages()) {
+                return discoverEndpointReport;
+            }
+            testUrl = discoverEndpointReport.getProbeResult(DISCOVERED_URL);
         }
-
-        Map<String, Object> probeResult = new HashMap<>();
-        probeResult.put(DISCOVERED_SOURCES,
-                createdConfig.get()
-                        .configurationHandlerId(WFS_SOURCE_CONFIGURATION_HANDLER_ID));
-
-        return probeReport.addMessage(buildMessage(SUCCESS_TYPES,
-                FAILURE_TYPES,
-                WARNING_TYPES,
-                CONFIG_CREATED))
-                .probeResults(probeResult);
+        return getPreferredWfsConfig(testUrl, configuration.sourceUserName(), configuration.sourceUserPassword());
     }
 
     @Override
     public List<ConfigurationMessage> validateOptionalFields(WfsSourceConfiguration configuration) {
-        return validateOptionalUsernameAndPassword(configuration);
+        List<ConfigurationMessage> results;
```

Comment:
We could move this to a method inside of `SourceValidationUtils` like we did with `validateOptionalUsernameAndPassword`. I think this is a good idea, anyone who makes a source is likely going to be reusing it.